embulk-input-s3/src/main/java/org/embulk/input/s3/HttpProxy.java

```diff
@@ -1,17 +1,39 @@
 package org.embulk.input.s3;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Optional;
+import org.embulk.config.Config;
+import org.embulk.config.ConfigDefault;
+import org.embulk.config.Task;
 
 /**
  * HttpProxy is config unit for Input/Output plugins' configs.
  *
  * TODO
  * This unit will be moved to embulk/embulk-plugin-units.git.
  */
-public class HttpProxy
+public interface HttpProxy
+    extends Task
 {
+    @Config("host")
+    public String getHost();
+
+    @Config("port")
+    @ConfigDefault("null")
+    public Optional<Integer> getPort();
+
+    @Config("https")
+    @ConfigDefault("true")
+    public boolean getHttps();
+
+    @Config("user")
+    @ConfigDefault("null")
+    public Optional<String> getUser();
+
+    @Config("password")
+    @ConfigDefault("null")
+    public Optional<String> getPassword();
+
+    /* TODO We can use this by jackson-core v2.6
```

Comment:
I don't think we should leave the following commented-out code in the repo. How about:
```
// TODO: Consider using @JsonProperty(defaultValue=...) in Jackson 2.6+.
```